cypress/e2e/with_api/catalogueCategory.cy.ts

```diff
@@ -0,0 +1,323 @@
+  modifyCatalogueCategory({
+    name: 'Lensess',
+  });
+
+  cy.findByText('Lensess').click();
+
+  modifyCatalogueCategory({
+    name: 'Spherical Lenses2',
+    newFormFields: [
+      {
+        name: 'Substrate',
+        type: 'text',
+        mandatory: true,
+        allowed_values: {
+          type: 'list',
+          values: ['N-BK7', 'UV Fused Siilica', 'Fused Silica'],
+        },
+      },
+      {
+        name: 'Diameter2',
+        type: 'number',
+        unit: 'mm',
+        mandatory: false,
+      },
+      {
+        name: 'Wavelength Range',
+        type: 'text',
+        unit: 'nm',
+        mandatory: true,
+      },
+      {
+        name: 'Broken3',
+        type: 'boolean',
+        mandatory: false,
+      },
+    ],
+  });
+};
+
+const editCatalogueCategories = () => {
+  cy.findByRole('button', { name: 'navigate to catalogue home' }).click();
+  modifyCatalogueCategory({
+    editCatalogueCategoryName: 'Lensess',
+    name: 'Lenses',
+  });
+
+  cy.findByText('Lenses').click();
+
+  modifyCatalogueCategory({
+    editCatalogueCategoryName: 'Spherical Lenses2',
+    name: 'Spherical Lenses',
+    newFormFields: [
+      {
+        name: 'Substrate',
+        type: 'text',
+        mandatory: true,
+        allowed_values: {
+          type: 'list',
+          values: ['N-BK7', 'UV Fused Silica', 'Fused Silica'],
+        },
+      },
+      {
+        name: 'Diameter',
+        type: 'number',
+        unit: 'mm',
+        mandatory: false,
+      },
+      {
+        name: 'Wavelength Range',
+        type: 'text',
+        unit: 'nm',
+        mandatory: true,
+      },
+      {
+        name: 'Broken',
+        type: 'boolean',
+        mandatory: false,
+      },
+    ],
+  });
+};
+
+const saveAsCatalogueCategories = () => {
+  cy.findByRole('button', { name: 'navigate to catalogue home' }).click();
+  saveAsCatalogueCategory('Lenses');
+  cy.findByText('Lenses').click();
+  saveAsCatalogueCategory('Spherical Lenses');
+};
+
+const copyToCatalogueCategories = () => {
+  copyToCatalogueCategory({
+    checkedCategories: ['Spherical Lenses', 'Spherical Lenses_copy_1'],
+  });
+};
+
+const moveToCatalogueCategories = () => {
+  cy.findByRole('button', { name: 'navigate to catalogue home' }).click();
+  cy.findByText('Lenses').click();
+  moveToCatalogueCategory({
+    checkedCategories: ['Spherical Lenses', 'Spherical Lenses_copy_1'],
+  });
+};
+
+const deleteCatalogueCategories = () => {
+  cy.findByRole('button', { name: 'navigate to catalogue home' }).click();
+  deleteCatalogueCategory('Spherical Lenses');
+  deleteCatalogueCategory('Spherical Lenses_copy_1');
+  deleteCatalogueCategory('Lenses');
+  deleteCatalogueCategory('Lenses_copy_1');
+};
+
+describe('Catalogue Category', () => {
+  beforeEach(() => {
+    cy.dropIMSDB();
+    cy.visit('/catalogue');
+  });
+  afterEach(() => {
+    cy.clearMocks();
+    cy.dropIMSDB();
+  });
+
+  it('CRUD for catalogue categories', () => {
```

Comment:
The only thing I think is missing is the behaviour when removing a field but considering we do just take the whole properties, I think this is alright for now.